frontend/src/components/auth/auth.tsx

```diff
@@ -2,21 +2,23 @@ import { FC } from 'react';
 import { MasterSignUpRequestDto, MasterSignInDto } from 'common/types/types';
 import { auth as authActions } from 'store/actions';
 import { AppRoute } from 'common/enums/enums';
-import { useLocation, useAppDispatch } from 'hooks/hooks';
+import { useLocation, useAppDispatch, useNavigate } from 'hooks/hooks';
 import logo from 'assets/img/logo.svg';
 import { SignInForm, SignUpForm } from './components/components';
 import styles from './auth.module.scss';
 
 const Auth: FC = () => {
   const { pathname } = useLocation();
   const dispatch = useAppDispatch();
+  const navigate = useNavigate();
 
   const handleSignInSubmit = (payload: MasterSignInDto): void => {
     dispatch(authActions.signIn(payload));
   };
 
   const handleSignUpSubmit = (payload: MasterSignUpRequestDto): void => {
     dispatch(authActions.signUp(payload));
+    navigate(AppRoute.ROOT, { replace: true });
```

Comment:
```suggestion
```
what to do if an error occurs?
we should do this logic in the action. Create a history service for it